api docs: Removed duplicacy in CSS.

MSurfer20 opened the pull request:
Deduplicate CSS for coloration of data types of response and
request parameters in API Documentation.

<!-- What's this PR for?  (Just a link to an issue is fine.) -->


**Testing plan:** <!-- How have you tested? -->


**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

showell commented:
@MSurfer20 Thanks for making this change.  I have a nitpick on the commit message.  "Duplicacy" is not actually a word.  You can either change it to "duplication" or just be more specific:

~~~
api docs: Cleaned up CSS for parameter classes.

I renamed (....) to (...) and added (...).
~~~

I am not sure I completely agree on the refactoring here, but I'll admit that I haven't worked closely with the API docs.

It seems like we would want some symmetry, so maybe `api-request-parameter-datatype` and `api-response-parameter-datatype`.  And then for any CSS that is common between them we could always do something like this:

~~~ css
.api-request-parameter-datatype,
.api-response-parameter-datatype {
  color: blue;
}

MSurfer20 commented:
Thanks for the review @showell ! I have changed the commit message, and changed the classes from inherited to separate ones as you suggested(even I initially thought to use the same approach of shared CSS, but didn't find many(any?) uses of this syntax in portico.css, but found many extends cases, so I had gone with it).

timabbott commented on static/styles/portico/portico.css:
```diff
@@ -1717,19 +1717,19 @@ label.label-title {
+    font-weight: 600;
+    font-size: 14px;
+}
+
```
Why not just have a single `.api-field-type` common class name used for rendering the data types in both contexts, without reference to request to response (since I imagine that detail doesn't affect the formatting)?  

Or is there a font size difference here you're trying to emphasize?  If so, I still think that'd be better as extra sizing CSS for the `.api-argument .api-field-type` selector.

timabbott replied:
(What I'm trying to get away from is parallel CSS with semi-duplicate names for a single concept).

MSurfer20 replied:
@timabbott Made the required changes

timabbott commented:
Merged, thanks @MSurfer20!  I'd next look at whether there's anything else we can refactor away from the `api-argument-blah` and `api-response-blah` to a common `api-field-blah` class.
